includes/options/class-amp-options-manager.php

```diff
@@ -319,4 +320,27 @@ public static function persistent_object_caching_notice() {
 			);
 		}
 	}
+
+	/**
+	 * Render the cache miss admin notice.
+	 *
+	 * @return void
+	 */
+	public static function render_cache_miss_notice() {
+		if ( 'toplevel_page_' . self::OPTION_NAME !== get_current_screen()->id ) {
+			return;
+		}
+
+		$cache_miss_url = get_option( AMP_Theme_Support::CACHE_MISS_URL_OPTION, false );
+		if ( empty( $cache_miss_url ) ) {
+			return;
+		}
+
+		printf(
+			'<div class="notice notice-warning"><p>%s <a href="%s">%s</a></p></div>',
+			esc_html__( 'Response caching was disabled due to exceeding the cache miss threshold.', 'amp' ),
```

Comment:
We'll probably need to figure out a way to explain what this means to non-developers. Maybe there should be a link to the Wiki with more information about what causes this? 

Also, should the notice be dismissible?

Should there be a checkbox to re-enable? If so, then the notice could be displayed inline with the checkbox.

Reply:
>We'll probably need to figure out a way to explain what this means to non-developers. Maybe there should be a link to the Wiki with more information about what causes this?

Agreed.  We need more work on the messaging.  I've noted it as a `@todo`.

>should the notice be dismissible?

I think so.  I think it's okay that the developer or admin dismisses this warning. 

Thinking through it, I'd like to use this top-level warning as an _alert_: "Hey, we automatically turned off caching. Go to the AMP > General page for more information."  Use it to alert them that we did something automatically and it needs their attention.

>Should there be a checkbox to re-enable? If so, then the notice could be displayed inline with the checkbox.

Agreed. This is the next step, i.e. adding a re-enable checkbox that appears when caching is disabled.